clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -239,9 +249,68 @@ private static FilterResult filterTo(TopicPartition partition, Iterable<MutableR
         return filterResult;
     }
 
+    private static BatchFilterResult filterBatch(RecordBatch batch,
+                                                 BufferSupplier decompressionBufferSupplier,
+                                                 FilterResult filterResult,
+                                                 RecordFilter filter,
+                                                 byte batchMagic,
+                                                 boolean writeOriginalBatch,
+                                                 long maxOffset,
```

Comment:
Looks like maxOffset is always initialized to -1?  If so, we can remove it from input parameters.

Reply:
thanks for the catch! I've moved it out of the input parameters to clean it up